File: tests/interact/marks_op.cpp
#include "../../include/optitrust.h"

typedef struct { int x; int y; } vect;

int f(int n) {
  for (int i = 0; i < 10; i++) {
    for (int j = 0; j < 12; j++) {
       i++;
       j++;
    }
  }
  return 3;
}

int main() {
  for (int i = 0; i < 3; i++) {
    vect r = { f(2), f(3) };
  }
}

File: tests/interact/marks_op.ml
open Optitrust
open Target

let _ = Run.script_cpp (fun _ ->

  !! Marks.add "M1" [nbMulti; cFor "i"];
  !! Marks.add "M2" [nbMulti; cFor "i"];
  !! Marks.add "M3" [nbMulti; cFunDef "main"; cFor "i"];
  !! Marks.remove "M1" [nbMulti; cFor "i"];
  !! Marks.add "M4" [nbMulti; cFor "i"];
  !! Marks.add "M5" [sInstr "i++"; tAfter];
  !! Marks.remove "M5" [cMark "M5"];
  !! Marks.clean [nbMulti; cFor "i"];
)
